pkg/flow/internal/controller/metrics.go

```diff
@@ -11,18 +13,20 @@ type controllerMetrics struct {
 }
 
 // newControllerMetrics inits the metrics for the components controller
-func newControllerMetrics(r prometheus.Registerer) *controllerMetrics {
+func newControllerMetrics(r prometheus.Registerer, id string) *controllerMetrics {
 	cm := controllerMetrics{r: r}
 
 	cm.controllerEvaluation = prometheus.NewGauge(prometheus.GaugeOpts{
-		Name: "agent_component_controller_evaluating",
-		Help: "Tracks if the controller is currently in the middle of a graph evaluation",
+		Name:        "agent_component_controller_evaluating",
+		Help:        "Tracks if the controller is currently in the middle of a graph evaluation",
+		ConstLabels: map[string]string{"controller_id": id},
 	})
 
 	cm.componentEvaluationTime = prometheus.NewHistogram(
 		prometheus.HistogramOpts{
-			Name: "agent_component_evaluation_seconds",
-			Help: "Time spent performing component evaluation",
+			Name:        "agent_component_evaluation_seconds",
+			Help:        "Time spent performing component evaluation",
+			ConstLabels: map[string]string{"controller_id": id},
 		},
```

Comment:
These metrics will leak now if a module terminates, since they're registered with MustRegister and never removed.